src/utils/tests-setup.ts

```diff
@@ -1,36 +1,13 @@
 import chai from "chai";
 import chaiAsPromised from "chai-as-promised";
 import type { Address } from "viem";
-import { BotEventNames } from "../bot-events";
-import EventEmitter from "node:events";
-import * as mocks from "./test-mocks";
+import { resolveOnEvent } from "./emitter";
 import { env } from "./env";
+import * as mocks from "./test-mocks";
 
 chai.use(chaiAsPromised);
 export const expect = chai.expect;
 
-/**
```

Comment:
This logic was initially only useful for tests, but it got extracted to use it in the application